renderer.go

```diff
@@ -1,38 +1,53 @@
 package lipgloss
 
 import (
-	"io"
+	"os"
 	"sync"
 
-	"github.com/muesli/termenv"
+	"github.com/charmbracelet/x/exp/term"
+	"github.com/lucasb-eyer/go-colorful"
 )
 
 // We're manually creating the struct here to avoid initializing the output and
 // query the terminal multiple times.
-var renderer = &Renderer{
-	output: termenv.DefaultOutput(),
-}
+var (
+	renderer     *Renderer
+	rendererOnce sync.Once
+)
 
 // Renderer is a lipgloss terminal renderer.
 type Renderer struct {
-	output            *termenv.Output
-	colorProfile      termenv.Profile
+	colorProfile      Profile
+	mtx               sync.RWMutex
 	hasDarkBackground bool
-
-	getColorProfile      sync.Once
-	explicitColorProfile bool
-
-	getBackgroundColor      sync.Once
-	explicitBackgroundColor bool
-
-	mtx sync.RWMutex
 }
 
-// RendererOption is a function that can be used to configure a [Renderer].
-type RendererOption func(r *Renderer)
-
 // DefaultRenderer returns the default renderer.
 func DefaultRenderer() *Renderer {
+	rendererOnce.Do(func() {
+		if renderer != nil {
+			// Alredy set by SetDefaultRenderer
+			return
+		}
```

Comment:
maybe instead do `rendererOnce.Do(func() { renderer = r })` on `SetRenderer`, then this would not be needed anymore 🤔